crypto/src/key/mod.rs

```diff
@@ -89,6 +89,13 @@ impl PrivateKey {
         };
         Ok(signature)
     }
+
+    /// Returns internal private key data (32-byte slice for Secp256k1Schnorr)
+    pub fn as_bytes(&self) -> &[u8] {
+        match &self.key {
+            PrivateKeyHolder::Secp256k1Schnorr(key) => key.as_bytes(),
+        }
+    }
```

Comment:
Why? This is wrong. We should strictly use the Encode trait if we need to serialize. The key details should be opaque.

Reply:
This is used to get raw bytes from the private key (32 bytes in the case of Secp256k1Schnorr). And I needed exactly 32 bytes to derive the VRF keys. We have now added SHA3, so we can now use serialized bytes as well.

Reply:
This is now gone and encode is used.